clients/src/main/java/org/apache/kafka/clients/producer/internals/ProduceRequestResult.java

```diff
@@ -41,12 +43,20 @@ public ProduceRequestResult() {
      * Mark this request as complete and unblock any threads waiting on its completion.
      * @param topicPartition The topic and partition to which this record set was sent was sent
      * @param baseOffset The base offset assigned to the record
+     * @param responseTimestamp The timestamp returned by the broker
      * @param error The error that occurred if there was one, or null.
      */
-    public void done(TopicPartition topicPartition, long baseOffset, RuntimeException error) {
+    public void set(TopicPartition topicPartition, long baseOffset, long responseTimestamp, RuntimeException error) {
         this.topicPartition = topicPartition;
         this.baseOffset = baseOffset;
+        this.responseTimestamp = responseTimestamp;
         this.error = error;
+    }
+
+    /**
+     * Mark this request as complete and unblock any threads waiting on its completion.
+     */
+    public void done() {
```

Comment:
Just one question for clarification: is it a requirement to call `set` before calling `done`? If so, would it make sense to add an assertion?